perllib/FixMyStreet/DB/Result/ModerationOriginalData.pm

```diff
@@ -139,26 +139,35 @@ sub compare_coords {
 sub compare_photo {
     my ($self, $other) = @_;
 
-    my $old = $self->photo || '';
-    my $new = $other->photo || '';
-    return '' if $old eq $new;
-
-    $old = [ split /,/, $old ];
-    $new = [ split /,/, $new ];
+    my $old = $self->photos;
+    my $new = $other->photos;
 
-    my $diff = Algorithm::Diff->new( $old, $new );
+    my (%old_photo_ids_set, %new_photo_ids_set);
     my (@added, @deleted);
-    while ( $diff->Next ) {
-        next if $diff->Same;
-        push @deleted, $diff->Items(1);
-        push @added, $diff->Items(2);
+
+    foreach (@$old) {
+        $old_photo_ids_set{$_->{id}} = 1;
+    }
+    foreach (@$new) {
+        $new_photo_ids_set{$_->{id}} = 1;
+        if (! exists $old_photo_ids_set{$_->{id}}) {
```

Comment:
You don't need `exists` because the value could only be undefined or `1` so a standard comparison is fine.
Perl has an `unless` which you can use as the opposite of `if` if that makes sense to do so, and for one-liner if/unless statements you can put the contents first, e.g.:
`push @added, $_ unless $old_photo_ids_set{$_->{id}};`
Which can sometimes read more clearly.